### PR TITLE
[#3004] - Allow disabling "Save current information" for forms.

### LINKED ISSUE #3004: Allow disabling "Save current information" for forms.
### Thema / Theme

Other

### Omschrijving / Description

"Save current information" is undesirable in some deployments of Open Forms. I propose adding a Boolean field `save_allowed` (default=`True`) on the `Form` model and exposing in the admin and `FormSerializer`.

- The SDK can then respond by not showing  the "Save current information" button.
- TBD: How to handle submission of data if save_allowed=`False`?

### Added value / Toegevoegde waarde

Some organizations don't want to use forms with features like this or may have other approaches to dealing with these flows. Making this configurable provides for this scenario.

### Aanvullende opmerkingen / Additional context

_No response_

### PR DESCRIPTION
**Pull request by svenvandescheur**
Depends on: https://github.com/open-formulieren/open-forms-sdk/pull/399

Closes #3004

### CODE REVIEW

File: src/openforms/submissions/api/serializers.py
Comment (sergei-maertens) [inline src/openforms/submissions/api/serializers.py:279]: this isn't right - this is not a field that the client may control. It needs to be looked up from the serializer instance for validation via `self.instance.form.suspend_allowed`

File: src/openforms/submissions/api/serializers.py
Comment (sergei-maertens) [inline src/openforms/submissions/api/serializers.py:298]: as discussed - this should raise a `ValidationError`, not a permission denied, if we bring it in the serializer instead of the view.

This is also not a serializer marked as experimental, so now I'm even more convinced that this should happen in the view instead of the serializer (where it should indeed raise a permission denied).

File: src/openforms/submissions/api/validation.py
Comment (sergei-maertens) [inline src/openforms/submissions/api/validation.py:37]:
```suggestion
    # TODO: Move permission check-like validators to view?
```

File: src/openforms/submissions/api/viewsets.py
Comment (sergei-maertens) [inline src/openforms/submissions/api/viewsets.py:353]: where is this change coming from?

File: src/openforms/submissions/api/viewsets.py
Comment (sergei-maertens) [inline src/openforms/submissions/api/viewsets.py:337]: this is the wrong approach, better would be to do:

```py
submission = self.get_object()
if not submission.form.submission_allowed:
    raise PermissionDenied(...)
serializer = SubmissionSuspensionSerializer(...)
```

File: src/openforms/submissions/api/serializers.py
Comment (CharString) [inline src/openforms/submissions/api/serializers.py:299]: Yeah, I think this check belongs in the `_suspend` view, right after loading the submission and before de-serialising the user input. The fact that it doesn't raise a `ValidationError` is a tell tail sign.

File: src/openforms/submissions/tests/test_submission_suspension.py
Comment (sergei-maertens) [inline src/openforms/submissions/tests/test_submission_suspension.py:81]:
```suggestion
        submission = SubmissionFactory.create(
            form__generate_minimal_setup=True,
            form__suspend_allowed=False,
        )
```

File: src/openforms/submissions/tests/test_submission_suspension.py
Comment (sergei-maertens) [inline src/openforms/submissions/tests/test_submission_suspension.py:98]:
```suggestion
        response = self.client.post(endpoint, {"email": "hello@open-forms.nl"})
        
        self.assertEqual(response.status_code, 403)
        error = response.json()["invalidParams"][0]
        self.assertEqual(error["code"], "permission_denied")
```

1. The error structure you're testing does not adhere to the documented error structure in the API spec
2. The status code and error code are most important, the error detail message is "fluff", no need to test for that and deal with translation machinery.

File: src/openforms/js/components/admin/form_design/FormConfigurationFields.js
Comment (sergei-maertens) [inline src/openforms/js/components/admin/form_design/FormConfigurationFields.js:314]: please add the default value of `True` to `form-creation-form.js`, in the `initialFormState` constant.

File: src/openforms/submissions/api/validation.py
Comment (CharString) [inline src/openforms/submissions/api/validation.py:38]: :thinking:  SubmissionAllowed is very much like the SuspensionAllowed. But accepting the privacy policy isn't. @sergei-maertens ?

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

**Review by CharString — DISMISSED** [review]
To keep the naming scheme the same as `submissionAllowed`, I'd use "suspension allowed" instead of "suspend allowed"

File: src/openforms/submissions/api/validation.py
Comment (sergei-maertens) [reply src/openforms/submissions/api/validation.py:38]: yeah we discussed this yesterday - cleaning this up right is out of scope (and the submit/complete serializer is still marked experimental so we can get away with it), but the privacy policy is something the client can fix/control, hence an HTTP 400 is appropriate, while the submission being blocked is more of a permission thing they have little control over, hence an HTTP 403 would be more appropriate.

HOWEVER - the steps being blocked CAN be derived from user input and logic, so a 400 may be appropriate if/when giving different answers unblocks submission. So it's a bit of a grey area :grimacing: Probably this should be broken up with the _complete endpoint checking if its outright blocked on the form -> HTTP 403 and otherwise if it's dynamically blocked -> HTTP 400 and leave it to the serializer.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
Can you please:

* Get rid of the merge migration by rebasing on master and re-generating the migration, so we get a linear migration history from new features

* Clean up your commit history by rebasing on master - there are now a number of commits without ticket reference and you can incorporate all the feedback in their relevant "feature" commits.

Other than that this looks fine to merge.

re: the migrations - we already end up with plenty of merge migrations from forward-ported fixes to older stable branches, so for the master branch we really want to keep a linear migration history as much as possible. You can roll back the migration on your dev machine first

**Comment by sergei-maertens** [comment]
I've cleaned up the commit history, should be good to merge now when CI completes.

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**
